packages/core/src/lib/lit-core.ts

```diff
@@ -645,31 +658,78 @@ export class LitCore {
     const serverKeys: Record<string, JsonHandshakeResponse> = {};
 
     let timeoutHandle: ReturnType<typeof setTimeout>;
+    let canRetry = this.config.retryNodeHandshake ?? true;
+
+    const handshakeWithNode = async (url: string): Promise<void> => {
+      serverKeys[url] = await this._handshakeAndVerifyNodeAttestation({
+        url,
+        requestId,
+      });
+      connectedNodes.add(url);
+    };
+
+    const handshakeWithNodeWithRetry = async (url: string): Promise<void> => {
+      try {
+        return await handshakeWithNode(url);
+      } catch (e) {
+        const errorInfo = VError.info(e);
+        // Only retry requests if there was a 5xx error or a timeout error (not client side errors)
+        const isRetryableRequest =
+          errorInfo.statusCode >= 500 ||
```

Comment:
we might also want to avoid retrying `404` `429` etc.

Reply:
It is avoiding those. It is flagging the request as retryable only for 5xx errros

Reply:
Ah I thought this was inverted. Thanks for the clarification.